#!/usr/bin/env python

PACKAGE = "omni_robot_odom"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

integration_method_enum = gen.enum([gen.const("Euler", int_t, 0, "Euler integration method"),
                                    gen.const("RK2", int_t, 1, "Runge-Kutta 2 integration method"),
                                    gen.const("RK4", int_t, 2, "Runge-Kutta 4 integration method")],
                                    "Enum for selecting integration method")

gen.add("integration_selector", int_t, 0, "Parameter for switching integration method", 0, edit_method = integration_method_enum)

exit(gen.generate(PACKAGE, "odom", "integration"))